### PR TITLE
Feat/Xline Handle

### PR DESCRIPTION
**Pull request by iGxnon**
Base on #27 

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: operator-api/src/xline.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            child_proc: None,
+        }
+    }
+}
+
+#[async_trait]
+impl XlineHandle for LocalXlineHandle {
+    async fn start(&mut self) -> anyhow::Result<()> {
+        self.kill().await?;
+        let mut cmds = self.start_cmd.split_whitespace();
```
Comment (bsbds) [inline operator-api/src/xline.rs:41]: Could we use Vec<String> as the type of `start_cmd` directly?

File: sidecar/src/types.rs [hunk 2]
```diff
@@ -75,7 +73,9 @@ impl Config {
     }
```
Comment (bsbds) [inline sidecar/src/types.rs:73]: Why the member name need to format to host:name?

File: sidecar/src/xline.rs [hunk 3]
```diff
@@ -88,17 +105,97 @@ impl XlineHandle {
     }
 
     /// Start the xline server
-    pub(crate) fn start(&self) -> Result<()> {
-        // TODO start xline server
+    pub(crate) async fn start(&mut self) -> Result<()> {
+        // TODO: hold a distributed lock during start
+
+        // Step 1: Check if there is any node running
+        // Step 2: If there is no node running, start single node cluster
```
Comment (bsbds) [inline sidecar/src/xline.rs:112]: Why start a single node cluster here, currently it seems Xline does not support membership change for a single node cluster, single node is designed for test only.

File: sidecar/src/types.rs (hunk 2 above)
Comment (iGxnon) [reply sidecar/src/types.rs:73]: How come?

File: sidecar/src/xline.rs (hunk 3 above)
Comment (iGxnon) [reply sidecar/src/xline.rs:112]: We should support it because etcd also supports it.

File: operator-api/src/xline.rs (hunk 1 above)
Comment (iGxnon) [reply operator-api/src/xline.rs:41]: It is refactored in the next PRs

File: sidecar/src/types.rs (hunk 2 above)
Comment (bsbds) [reply sidecar/src/types.rs:73]: sorry I read the code wrong

**Review by bsbds — APPROVED** [review]

**Pull request closed without merging**